pkg/controller/configmap/config.go

```diff
@@ -469,11 +470,35 @@ func NeedApplyChange(currConfig *corev1.ConfigMap, prevConfig *corev1.ConfigMap)
 		_, err = prevCfg.GetSection(name)
 		if err != nil {
 			if len(currCfg.Section(name).KeyStrings()) != 0 {
+				log.Info(fmt.Sprintf("Section [%s] is added into configmap", name))
 				diffSecs = append(diffSecs, name)
 			}
 			continue
 		}
 		if !reflect.DeepEqual(currCfg.Section(name).KeysHash(), prevCfg.Section(name).KeysHash()) {
+			// Check which configmap option value is changed
+			keys := currCfg.Section(name).KeyStrings()
+			for _, key := range keys {
+				curKeyValue := currCfg.Section(name).Key(key).Value()
+				if prevCfg.Section(name).HasKey(key) {
+					preKeyValue := prevCfg.Section(name).Key(key).Value()
+					if !reflect.DeepEqual(preKeyValue, curKeyValue) {
+						log.Info(fmt.Sprintf("Section [%s] config option: %s = %s is changed to %s = %s",
+							name, key, preKeyValue, key, curKeyValue))
+					}
+				} else {
+					log.Info(fmt.Sprintf("Section [%s] add/uncomment a new config option: %s = %s",
+						name, key, curKeyValue))
```

Comment:
Do you think we should hide the value as maybe it's related to cert/password?

Reply:
From debug perspective, we want to show some option value's in the log for better understanding what changes have been done. We might change it to debug level log. but printing value itself in somehow is worthy. Or we can
do special handing for cert/password to hide value. That's maybe better.